src/modules/formBuilder/components/FormBuilder.tsx

```diff
@@ -107,11 +108,96 @@ const FormBuilder: React.FC<FormBuilderProps> = ({
+        />
+      )}
+      <Box display='flex'>
+        <Box sx={{ flexGrow: 1 }}>
+          <Box
+            sx={{
+              // Matches the styles usually applied in DashboardContentContainer.
+              // (Moved in here because of the Palette drawer)
+              maxWidth: `${theme.breakpoints.values.xl}px`,
+              pt: 2,
+              pb: 8,
+              px: { xs: 1, sm: 3, lg: 4 },
+            }}
+          >
+            <FormBuilderHeader
+              formDefinition={formDefinition}
+              lastUpdatedDate={lastUpdatedDate}
+              onClickPreview={onClickPreview}
+            />
+            <Box sx={{ p: 4 }}>
+              <FormTree
+                definition={workingDefinition}
+                onEditClick={(item: FormItem) => {
+                  function editItem() {
+                    setSelectedItem(item);
+                  }
+
+                  if (dirty) {
+                    // React's useState accepts either a value or a function that yields a value.
+                    // In this case, we want the function itself to *be* the state value, which is the reason
+                    // for defining `editItem` above instead of simply using `() => setSelectedItem(item)` here.
+                    setBlockedActionFunction(() => editItem);
+                  } else {
+                    setSelectedItem(item);
+                  }
+                }}
+              />
+              {errorState?.errors &&
+                errorState.errors.length > 0 &&
+                !selectedItem && (
+                  <Stack gap={1} sx={{ mt: 4 }}>
+                    <ErrorAlert key='errors' errors={errorState.errors} />
+                  </Stack>
+                )}
+            </Box>
+            {dirty && (
+              <Paper sx={{ p: 4 }}>
+                <Stack
+                  direction='row'
+                  justifyContent='space-between'
+                  sx={{ alignItems: 'center' }}
+                >
+                  <Stack direction='row' gap={2}>
+                    <LoadingButton
+                      variant='outlined'
+                      loading={saveLoading}
+                      onClick={() => onSave(workingDefinition)}
+                    >
+                      Save Draft
+                    </LoadingButton>
+                    <Button>Publish</Button>
+                  </Stack>
+                  <Typography variant='body2'>
+                    Last saved on {lastUpdatedDate} by {lastUpdatedBy}
+                  </Typography>
+                </Stack>
+              </Paper>
+            )}
+          </Box>
+        </Box>
         <FormBuilderPalette
           onItemClick={(itemType) => {
             const newItem: FormItem = {
-              linkId: crypto.randomUUID(), // TODO(#6083) - this is a placeholder, but right now will only work in localhost
+              linkId: crypto.randomUUID(), // TODO(#6083) - this is a placeholder
```

Comment:
I'd like to discuss this. I was wrong in my comment that it'd only work in localhost -- seems it works as long as the context is secure (https). However, these randomly generated UUIDs aren't compliant with our new link ID requirements because they have hyphens.

In general, do we want to take the approach of randomly generating link IDs? Or should we try to programmatically guess at a more human-readable link ID?

Reply:
All good questions! I think generating a random link id initially is appropriate, since we have nothing else to go off. Lets use the uuid library instead of crypto though:
```
import { v4 } from 'uuid';
v4() // remove dashes, or just take the first part `v4().split('-')[0]`
```

It would be even nicer if we replaced it with a readable LinkID once a user typed in a label. The advantage of readable Link IDs is that when they are referenced in other conditionals,  you can tell at a glance what they're talking about (when looking at the JSON). That might make it worth implementing.


I don't think it would be _too_ tricky to implement, now that we integrated RHF. We could use an onBlur on the label element to update the link id value. It probably shouldn't re-set itself if it was manually updated, which we should be able to tell from the dirtyFields state. The Link ID should probably be hidden in some "Advanced" section, too, since the average user shouldn't need to see it.

I fiddled around with a simple slugifier below. We'll likely want something similar/same for generating custom data element definition `key`s too.
```
linkId = kebabCase(item.label)
    .toLowerCase()
    .replace(/\-/g, '_')
    .slice(0, 40)
// if generated linkId not unique in the form, append unique uuid to it
```

What do you think of that approach?
If it's feeling too scope-creeping for this ticket, we can just go with uuid to start.